### ISSUE TITLE
KeyError when running 'badread error_model'

### ISSUE THREAD

**Issue by stcoh** [rrwick/Badread#33]
**Describe the bug**
Encountered error when generating my own error profile according to https://github.com/rrwick/Badread/wiki/Generating-error-and-qscore-models using https://downloads.pacbcloud.com/public/revio/2022Q4/HG002-rep1/.

**To reproduce**
```
SAMtools v1.14
samtools fastq -1 HG002-rep1_revio_2022Q4.R1.fastq.gz -2 HG002-rep1_revio_2022Q4.R2.fastq.gz -0 HG002-rep1_revio_2022Q4.fastq.gz -s HG002-rep1_revio_2022Q4.singleton.fastq.gz -N m84011_220902_175841_s1.hifi_reads.bam

minimap2 v2.26-r1175:
minimap2 -t 20 -c -x map-hifi hg38.fa HG002-rep1_revio_2022Q4.fastq.gz | gzip > alignments.paf.gz

Badread v0.4.0:
badread error_model --reference hg38.fa --reads HG002-rep1_revio_2022Q4.fastq.gz --alignment alignments.paf.gz > new_error_model
Loading reads......
Loading alignments......
Choosing best alignment per read......
Processing alignments........................................................................................................................................................................................................Traceback (most recent call last):
  File "/home/username/anaconda3/envs/visorenv/bin/badread", line 8, in <module>
    sys.exit(main())
  File "/home/username/anaconda3/envs/visorenv/lib/python3.8/site-packages/badread/__main__.py", line 38, in main
    make_error_model(args, output=output)
  File "/home/username/anaconda3/envs/visorenv/lib/python3.8/site-packages/badread/error_model.py", line 62, in make_error_model
    kmer_alternatives[ref_kmer][read_kmer] += 1
KeyError: 'CNTCAGG'
```

**Error message**
Traceback (most recent call last):
  File "/home/username/anaconda3/envs/visorenv/bin/badread", line 8, in <module>
    sys.exit(main())
  File "/home/username/anaconda3/envs/visorenv/lib/python3.8/site-packages/badread/__main__.py", line 38, in main
    make_error_model(args, output=output)
  File "/home/username/anaconda3/envs/visorenv/lib/python3.8/site-packages/badread/error_model.py", line 62, in make_error_model
    kmer_alternatives[ref_kmer][read_kmer] += 1
KeyError: 'CNTCAGG'

**System info**
 - Python v3.8.19

**Additional context**
Checked the reference, and KeyError: 'CNTCAGG' is somewhere in chr3. See below for context.
ATTAGAAAGTACATACCTCCTATAATTACTTGCTTGGGCAATAAACTAGA
AGTTAGTAAATATTAAGAACTTCAGGGTCCGGgcctgtaatcccagcatg
ttgggaggctgaggtgggggcggcgggggggggggggtggtgaatcacct
gaagtcaggagtttgagaccagcctgggcaacacggcaaaactccatctc
tactgataatacaaaaaaaaaatagctgggcgtggtggcgggcacctgta
atcccagctactagggaggctgaggcagaagaatcacttaaacgtgggag
gcggaggttgcagtgagctgagatgctgccactgcactccagcctgggca
**acagaggaaggctccatctcaaaaacaaaaacaaaaaaaaCNTCAGGGTC**
ATTACCCTATTTCTTCAAtttttaatttattttttatttttgagacagag
tctcactctgtcatccaggctggagtgcaatggcacaatctcagctcact
gcaatctctccctcccaggttcaagtgattctcctgcctcagcctctcaa
gtagctgggactacaggtacctgccaccatgcccagctaattttttattt
tttgtagagacagggtttcaccatgttggccaggctggtctcgaactcct
gacctcaagcaatctgcctgccttggcctcccaaagtgctgggattacag

**Comment by rrwick** [comment]
The problem is because Badread was only expecting A, C, G and T in the reference, so the N was a problem. But this should be fixed in a recent commit: fc1bd9821df478e981c614d21b775d75fc9636ab.

However, I haven't made a new release of Badread since I did that fix, so you'll need to [install from source](https://github.com/rrwick/Badread?tab=readme-ov-file#install-from-source). And it will be in the next release (whenever that is).

**Comment by stcoh** [comment]
> The problem is because Badread was only expecting A, C, G and T in the reference, so the N was a problem. But this should be fixed in a recent commit: [fc1bd98](https://github.com/rrwick/Badread/commit/fc1bd9821df478e981c614d21b775d75fc9636ab).
> 
> However, I haven't made a new release of Badread since I did that fix, so you'll need to [install from source](https://github.com/rrwick/Badread?tab=readme-ov-file#install-from-source). And it will be in the next release (whenever that is).

Thank you so much! It works out perfectly now.
